tools/lib/traverse.js

```diff
@@ -146,6 +146,24 @@ export class TraverseContext {
     if (!returns_async) {
       return undefined;
     }
+
+    /** @type {chromeTypes.TypeSpec} */
+    let callbackParameter = {
+      ...returns_async,
+      type: 'function',
+    };
+    // If this signature doesn't support promises we just convert the "returns_async" field to a
+    // callback.
+    if (returns_async.does_not_support_promises) {
+      return {
+        withCallback: {
```

Comment:
One last thing - isn't the callback optional even in this case? It does mean you don't have a way of knowing when the operation completes but if you don't mind about that I think it works.

Reply:
That makes sense, thanks for clarifying.